Add inelastic model for Rebar

SudiptaBiswas opened the pull request:
Addresses #74

moosebuild commented:
Job [Test](https://civet.inl.gov/job/405644/) on c3e9634 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/blackbear/docs/PRs/75/site)

This comment will be updated on new commits.

bwspenc commented on doc/content/source/materials/InelasticRebar.md:
```diff
@@ -0,0 +1,65 @@
+# Inelastic Rebar
```
I'd suggest renaming this to PlasticRebar (because plasticity is the only form of inelasticity that it represents).

bwspenc replied:
Actually, call it PlasticTruss (to be consistent with LinearElasticTruss), which exists already.

bwspenc commented on doc/content/source/materials/InelasticRebar.md:
```diff
@@ -0,0 +1,65 @@
+
+## Description
+
+The `InelasticRebar` model implements the J2 plasticity for reinforcement bars in concrete that are modeled using 1D truss elements. The model utilizes a discrete finite element implementation strategy to simplify the computational effort. The axial elongation of the element is obtained from
```
implements the J2 -> implements J2

I'd delete this sentence: The model utilizes a discrete finite element implementation strategy to simplify the computational effort. (it's implied in all of our models)

bwspenc commented on doc/content/source/materials/InelasticRebar.md:
```diff
@@ -0,0 +1,65 @@
+ \begin{equation}
+\sigma_a = E \frac{\Delta l}{l_0} \, .
+\end{equation}
+The nonlinear behavior of the reinforcement bar is implemented using a simplistic J2 plasticity model with linear hardening. The strain increment is obtained from the total stretch update
```
simplistic -> simple

bwspenc commented on include/materials/InelasticRebar.h:
```diff
@@ -0,0 +1,63 @@
+
+  /// maximum no. of iterations and iteration number
+  const unsigned int _max_its;
+  unsigned int _iteration;
```
_iteration shouldn't be a member variable.  Just decleare it as a local variable in the function where it is used.

bwspenc commented on include/materials/InelasticRebar.h:
```diff
@@ -0,0 +1,63 @@
+  unsigned int _iteration;
+
+  /// Residual values
+  Real _residual;
```
_residual also should be made a local variable.

bwspenc commented on test/tests/rebar/truss_inelastic.i:
```diff
@@ -0,0 +1,126 @@
+[Kernels]
+  [./solid]
+    type = StressDivergenceTensorsTruss
+    displacements = 'disp_x'
```
Put 'displacements' in GlobalParams and remove it in the 4 places it shows up in various models.

SudiptaBiswas commented:
I have addressed the comments and created a new PR in MOOSE https://github.com/idaholab/moose/pull/14310 contributing the codes to tensor_mechanics. Hence, closing this PR.
